lib/procore/resources/incidents.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Procore.Resources.Incidents do
+  @moduledoc """
+  Available actions for the Incident resource.
+  """
+
+  alias Procore.ErrorResult
+  alias Procore.Request
+  alias Procore.ResponseResult
+
+  @doc """
+  Lists all Incidents in a Project.
+  """
+  @spec list(Tesla.Client.t(), %{
+          (project_id :: String.t()) => pos_integer
+        }) :: %ResponseResult{} | %ErrorResult{}
+  def list(client, %{"project_id" => project_id} = params) do
+    %Request{}
+    |> Request.insert_request_type(:get)
+    |> Request.insert_endpoint("/vapid/projects/#{project_id}/incidents")
+    |> Request.insert_query_params(params)
```

Comment:
Do we need to be passing in the `project_id` as a query param here? The [docs](https://developers.procore.com/reference/incidents#list-incidents) only show it as a path param.

I think it would be similar to the list call for inspection types:
https://github.com/procore/elixir-sdk/blob/master/lib/procore/resources/inspection_types.ex#L10-L20

Reply:
Nope! All `project_id` params are within the path. 

Would I be able to define params as an empty collection, (in order to take in `view` and `filter` params), or would I have to explicitly define every possible valid param?

Reply:
It doesn't matter too much to me actually. I had forgotten about the possibility of view/filter params that could be introduced. I think either way works. I generally like to define all the possible params via pattern matching, but I do see the downside to it. And the other way has also been implemented already, https://github.com/procore/elixir-sdk/blob/master/lib/procore/resources/rfis.ex#L24-L43. Go ahead with what you think works best here.